### PR TITLE
Use errgroup

### PR DESCRIPTION
**Pull request by shashwathi**
## Proposed Changes

  * Use sync wait group in build controller

### CODE REVIEW

File: cmd/controller/main.go [hunk 1]
```diff
@@ -134,18 +135,19 @@ func main() {
 			logger.Fatalf("failed to wait for cache at index %v to sync", i)
 		}
 	}
-
+	var wg sync.WaitGroup
```
Comment (ImJasonH) [inline cmd/controller/main.go]: We could use https://godoc.org/golang.org/x/sync/errgroup here too, and just have each `g.Go` call `ctrlr.Run` and return its error.

```
var g errgroup.Group
for _, c := range controllers {
  c := c // capture c for the goroutine
  g.Go(func() error {
    return c.Run(...)
  })
}
if err := g.Wait(); err != nil {
  logger.Fatalf("...", err)
}
```

Comment (shashwathi) [reply cmd/controller/main.go]: I have never used `errgroup` before. That looks great

File: cmd/controller/main.go [hunk 2]
```diff
@@ -135,19 +135,20 @@ func main() {
...
-			if err := ctrlr.Run(threadsPerController, stopCh); err != nil {
-				logger.Fatalf("Error running controller: %s", err.Error())
-			}
-		}(ctrlr)
+			return ctrlr.Run(threadsPerController, stopCh)
+		})
+	}
+
+	// Wait for all controllers to finish and log errors if there are any
+	if err := g.Wait(); err == nil {
```
Comment (ImJasonH) [inline cmd/controller/main.go]: `err != nil`

File: cmd/controller/main.go [hunk 3]
```diff
@@ -134,18 +135,20 @@ func main() {
 			logger.Fatalf("failed to wait for cache at index %v to sync", i)
 		}
 	}
+	var g errgroup.Group
 
 	// Start all of the controllers.
 	for _, ctrlr := range controllers {
-		go func(ctrlr controller.Interface) {
+		ctrlr := ctrlr
```
Comment (pivotal-nader-ziada) [inline cmd/controller/main.go]: what does this line do?

Comment (pivotal-nader-ziada) [reply cmd/controller/main.go]: oh got it, maybe if the name was different, it would not be confusing

**Pull request merged**